src/FormParserFactory.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace React\Http;
+
+use React\Http\Parser\FormUrlencoded;
+use React\Http\Parser\Multipart;
+
+class FormParserFactory
+{
+    public static function create(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        if (!array_key_exists('content-type', $headers)) {
```

Comment:
Given the current design, it probably makes sense to throw an `Exception` here, because it can not actually be "parsed as a form".

I'm not sure I particularly like the current design, however.